integreat_cms/cms/models/chat/user_chat.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+import time
+from typing import TYPE_CHECKING
+
+from django.conf import settings
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from ..abstract_base_model import AbstractBaseModel
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+# pylint: disable=too-few-public-methods
+class UserChatManager(models.Manager):
+    """
+    custom manager providing function to get the current chat
+    """
+
+    def current_chat(self, device_id: str, **kwargs: Any) -> UserChat:
+        r"""
+        Return only the newest (i.e. current) chat for a given device_id
+
+        :param device_id: the device id for which we want the current chat
+        :param \**kwargs: The supplied kwargs
+        :return: the current chat for the given device_id
+        """
+        return (
+            self.get_queryset()
+            .filter(device_id=device_id, **kwargs)
+            .order_by("-pk")
+            .first()
+        )
+
+
+class UserChat(AbstractBaseModel):
+    """
+    A model for a user (app) chat, mapping a device ID to a Zammad ticket ID
+    """
+
+    device_id = models.CharField(max_length=200)
+    zammad_id = models.IntegerField()
```

Comment:
Just another design note for @steffenkleinle : one app can have multiple sessions, as the device_id must not be unique. In theory the app could provide an option to switch between multiple tickets.

Reply:
@steffenkleinle yes, but creating a new chat from the same device is already supported in this implementation.